src/components/MoneyReportHeader.tsx

```diff
@@ -108,8 +108,14 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
     const transactionIDs = allTransactions.map((t) => t.transactionID);
     const allHavePendingRTERViolation = TransactionUtils.allHavePendingRTERViolation(transactionIDs);
     const hasOnlyHeldExpenses = ReportUtils.hasOnlyHeldExpenses(moneyRequestReport.reportID);
+    const isPayAtEndExpense = TransactionUtils.isPayAtEndExpense(transaction);
+    const isArchivedReport = ReportUtils.isArchivedRoomWithID(moneyRequestReport?.reportID);
+    const [archiveReason] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${moneyRequestReport.reportID}`, {selector: ReportUtils.getArchiveReason});
 
-    const shouldShowPayButton = useMemo(() => IOU.canIOUBePaid(moneyRequestReport, chatReport, policy), [moneyRequestReport, chatReport, policy]);
+    const shouldShowPayButton = useMemo(
+        () => IOU.canIOUBePaid(moneyRequestReport, chatReport, policy, transaction ? [transaction] : undefined),
```

Comment:
NAB: but I don't see any expensive operations in `IOU.canIOUBePaid` so I think this `useMemo` call is unnecessary.

Reply:
I guess there is no harm in keeping it either? Besides this will prevent unnecessary re-renders, won't it?

Reply:
`useMemo` just re-uses the previously calculated value during re-renders so it doesn't really prevent re-renders on its own.

It's a performance optimization, but it's not "free" since it costs some memory to save the previous values of the dependencies and it also needs to compare those values to the new values. So in some cases, `useMemo` is actually _slower_ than just re-running the function directly. I'm not sure it is in this case, but I see no obviously slow operation in `canIOUBePaid` that should be memoized.

Reply:
That makes sense, learning a lot on this PR, thanks!

I don't feel super comfortable removing the `useMemo` that was previously there though, but if have strong feelings on this one I can remove it.

Reply:
Sure we'll leave it for now